ray-operator/controllers/ray/raycluster_controller_test.go

```diff
@@ -929,9 +941,56 @@ var _ = Context("Inside the default namespace", func() {
 
 		It("Check the number of worker Pods", func() {
 			numWorkerPods := 3 * int(numOfHosts)
+			desiredWorkerPods := replicas * numOfHosts
+			minWorkerPods := minReplicas * numOfHosts
+			maxWorkerPods := maxReplicas * numOfHosts
+
 			Eventually(
 				listResourceFunc(ctx, &workerPods, workerFilters...),
 				time.Second*3, time.Millisecond*500).Should(Equal(numWorkerPods), fmt.Sprintf("workerGroup %v", workerPods.Items))
+
+			Eventually(func() bool {
+				// "Replica" in status fields (e.g., DesiredWorkerReplicas) refers to the number of pods.
+				// The name is preserved for compatibility, but it semantically means pod count.
+				status := getClusterStatus(ctx, namespace, rayCluster.Name)()
+				if status.DesiredWorkerReplicas != desiredWorkerPods ||
+					status.MinWorkerReplicas != minWorkerPods ||
+					status.MaxWorkerReplicas != maxWorkerPods {
+					return false
+				}
+				return true
+			}, time.Second*5, time.Millisecond*500).Should(BeTrue())
+		})
+
+		It("Create a head Pod", func() {
+			err := k8sClient.List(ctx, &headPods, headFilters...)
+			Expect(err).NotTo(HaveOccurred(), "Failed to list head Pods")
+			Expect(headPods.Items).Should(HaveLen(1), "headPods: %v", headPods.Items)
+		})
+
+		It("Update all Pods to Running", func() {
+			// Note that this test assumes that headPods and workerPods are up-to-date.
+			for _, headPod := range headPods.Items {
+				headPod.Status.Phase = corev1.PodRunning
+				Expect(k8sClient.Status().Update(ctx, &headPod)).Should(Succeed())
+			}
+
+			Eventually(
+				isAllPodsRunningByFilters).WithContext(ctx).WithArguments(headPods, headFilters).WithTimeout(time.Second*3).WithPolling(time.Millisecond*500).Should(BeTrue(), "Head Pod should be running.")
+
+			for _, workerPod := range workerPods.Items {
+				workerPod.Status.Phase = corev1.PodRunning
+				Expect(k8sClient.Status().Update(ctx, &workerPod)).Should(Succeed())
+			}
+
+			Eventually(
+				isAllPodsRunningByFilters).WithContext(ctx).WithArguments(workerPods, workerFilters).WithTimeout(time.Second*3).WithPolling(time.Millisecond*500).Should(BeTrue(), "All worker Pods should be running.")
+		})
+
+		It("RayCluster's .status.state should be updated to 'ready' shortly after all Pods are Running", func() {
```

Comment:
Move the checks for `DesiredWorkerReplicas`, `MinWorkerReplicas`, and `MaxWorkerReplicas` to this block, and rename this block.